app/controllers/application_controller.rb

```diff
@@ -44,36 +43,29 @@ def create_user_event(event_type, user = current_user)
     Event.create(user_id: user.id, event_type: event_type)
   end
 
+  def decorated_session
+    @_decorated_session ||= DecoratedSession.new(sp: current_sp, view_context: view_context).call
+  end
+
   private
 
   def redirect_on_timeout
-    params = request.query_parameters
     return unless params[:timeout]
 
-    params[:issuer].present? ? redirect_with_sp : redirect_without_sp
-  end
-
-  def sp_metadata
-    ServiceProvider.from_issuer(request.query_parameters[:issuer]).metadata
+    flash[:timeout] = decorated_session.timeout_flash_text
```

Comment:
@hursey013 This is what I ended up doing to eliminate the Rubocop offenses:

- move the flash message text to the existing session decorators
- eliminate the need for methods like `sp_metadata` and `sp_name` in `ApplicationController` by reading the SP attributes from the DB via `current_sp` and `decorated_session`, which fetch the SP from the DB using the `issuer` provided by either `sp_session` or `params`.